tests/rest_api/test_0100_jobs.py

```diff
@@ -0,0 +1,170 @@
+        jobs, kwargs = filter_jobs(jobs, tasks, org)
+        self._test_list_jobs_200('admin1', jobs, **kwargs)
+
+    @pytest.mark.parametrize('org_id', ['', None, 1, 2])
+    @pytest.mark.parametrize('groups', [['business'], ['user'], ['worker'], []])
+    def test_non_admin_list_jobs(self, org_id, groups, users, jobs, tasks,
+        projects, memberships):
+        # keep the reasonable amount of users and jobs
+        users = [u for u in users if u['groups'] == groups][:2]
+        jobs, kwargs = filter_jobs(jobs, tasks, org_id)
+        org_staff = get_org_staff(org_id, memberships)
+
+        for user in users:
+            user_jobs = []
+            for job in jobs:
+                job_staff = get_job_staff(job, tasks, projects)
+                if user['id'] in job_staff | org_staff:
+                    user_jobs.append(job)
+            if is_org_member(memberships, user, org_id):
+                self._test_list_jobs_200(user['username'], user_jobs, **kwargs)
+            else:
+                self._test_list_jobs_403(user['username'], **kwargs)
+
+
+class TestGetAnnotations:
+    def _test_get_job_annotations_200(self, user, jid, data, **kwargs):
+        response = get_method(user, f'jobs/{jid}/annotations', **kwargs)
+
+        assert response.status_code == HTTPStatus.OK
+        assert DeepDiff(data, response.json()) == {}
+
+    def _test_get_job_annotations_403(self, user, jid, **kwargs):
+        response = get_method(user, f'jobs/{jid}/annotations', **kwargs)
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    @pytest.mark.parametrize('org', [None, '', 1, 2])
+    def test_admin_get_job_annotations(self, jobs, tasks, annotations, org):
+        jobs, kwargs = filter_jobs(jobs, tasks, org)
+
+        # keep only the reasonable amount of jobs
+        for job in jobs[:8]:
+            jid = str(job['id'])
+            self._test_get_job_annotations_200('admin2', jid,
+                annotations['job'][jid], **kwargs)
+
+    @pytest.mark.parametrize('org_id', ['', None])
+    @pytest.mark.parametrize('groups', [['business'], ['user'], ['worker'], []])
+    def test_non_admin_get_job_annotations(self, org_id, groups, users, jobs, tasks,
+        projects, annotations, memberships):
+        users = [u for u in users if u['groups'] == groups][:4]
+        jobs, kwargs = filter_jobs(jobs, tasks, org_id)
+        org_staff = get_org_staff(org_id, memberships)
+
+        # keep only the reasonable amount of jobs
+        for job in jobs[:8]:
+            job_staff = get_job_staff(job, tasks, projects)
+            jid = str(job['id'])
+
+            for user in users:
+                if user['id'] in job_staff | org_staff:
+                    self._test_get_job_annotations_200(user['username'],
+                        jid, annotations['job'][jid], **kwargs)
+                else:
+                    self._test_get_job_annotations_403(user['username'],
+                        jid, **kwargs)
```

Comment:
I see that this test can do more than one test for a similar test case, for example for test case (worker, not job staff) the function `_test_get_job_annotations_403` will be call more than once. I think from IAM testing point of view we have no reason in testing similar action for all users with same privilege/memberships. What do you think, @nmanovic?

Reply:
@kirill-sizov , I agree. If you see an easy way to reduce redundancy, I'm OK with the approach.